### DISCUSSION TITLE
Distribute to flathub

### DISCUSSION THREAD

**Discussion by hrzlgnm** [hrzlgnm/mdns-browser#2447] [General]
https://v2.tauri.app/distribute/flatpak/

**Comment by jing2uo** [comment]
Hi @hrzlgnm — thanks for mdns-browser, it's become my go-to for looking at what's advertising itself on a network.

I maintain [FlatPark](https://github.com/flatpark/flatpark), a small hub that provides Flatpak installs for desktop apps that don't have one yet, and I've packaged mdns-browser there, so the option exists for people who want it with no work on your side.

It uses Flatpak **extra-data**, so it downloads your official `mdns-browser_<version>_amd64.deb` unmodified at install time and tracks new releases automatically from your GitHub releases — I don't rehost or patch anything.

Install:

```
flatpak remote-add --if-not-exists flatpark https://dl.flatpark.org/flatpark.flatpakrepo
flatpak install flatpark io.github.hrzlgnm.mdns-browser
```

App page: https://flatpark.org/apps/io.github.hrzlgnm.mdns-browser

Three notes on how the package is configured, in case they're ever useful to you:

- `--share=network` is sufficient for discovery. It puts the app in the host network namespace, so multicast on 224.0.0.251:5353 reaches it normally; no extra capability is needed.
- The package needs no filesystem access at all, so it ships with none.
- While writing the update resolver I noticed the release tags moved from `mdns-browser-v1.9.19` to `v1.15.0`, so I read the asset URL from the release's asset list rather than composing it from the tag. Mentioning it only in case it matters for anything else consuming your releases.

I've only smoke-tested it myself — it installs, launches and renders correctly with no missing libraries — but I have not confirmed that services actually populate the list on a real LAN, so my coverage is certainly incomplete. If you have time to put it through real use and see what breaks, I'd genuinely appreciate it, and any issue or PR on the packaging is very welcome.

If you'd be willing to make FlatPark an officially supported install method — recommending it in your README, docs or download page — I would be very grateful. It takes the Flatpak packaging and distribution work off your plate while giving your users a clean install and automatic updates.

And if you'd rather I not distribute it, just say so and I'll take it down right away, no questions asked. If you're happy for it to stay, I'll add a blue "developer-approved" shield to its FlatPark page and feature it on the homepage.

**Reply by hrzlgnm** [reply]
I also did some testing on a ubuntu 26.04 machine, and browsing seems to work.

**Comment by hrzlgnm** [comment]
@jing2uo Thanks for reaching out. I appreciate the addition to FlatPark. I'll add it to the readme as an official install methods, but some minor issues need to be addressed first:
* You are using the executable from the .deb, which has auto-updates enabled, which may be confusing for users as it may or may not work at all or install the deb next to the flatpack version. Please consider using the so called un-bundled executable instead, which is also provided as a release asset, as an example for [v1.15.0](https://github.com/hrzlgnm/mdns-browser/releases/download/v1.15.0/mdns-browser_linux_x64). This executable does not provide the auto-update capability and is more suitable for re-packaging and comes from the same build before the `bundle tags` are patched into the executable. 
* You are setting  WEBKIT_DISABLE_DMABUF_RENDERER=1 in your wrapper script, which is normally not needed, as mdns-browser auto-detects whether those workarounds are needed for X11/wayland and enables those itself. But therefore it needs access to /sys/module/* and several other system files which may not be accessible in the flatpak sandbox. I need to verify first whether the detection works properly inside the flatpak sandbox, so you can keep that for now.

**Comment by hrzlgnm** [comment]
@jing2uo in the upcoming v1.15.2, the webkit2gtk workaround detection should also work in a flatpak environment reliably, i've updated the respective quirk to be more flatpak compatible. So the explicit workaround with setting WEBKIT_DISABLE_DMABUF_RENDERER=1 should not be needed anymore.

**Comment by hrzlgnm** [comment]
FlatPark installation method is added to readme with https://github.com/hrzlgnm/mdns-browser/pull/2449

**Comment by hrzlgnm** [comment]
Update on the FlatPark packaging: the Flatpak now uses the unbundled `mdns-browser_linux_x64` executable as you suggested, instead of the executable from the `.deb` — see flatpark/flatpark#197. Thanks again for the guidance!

**Reply by jing2uo** [reply]
Thank you, @hrzlgnm — this means a lot to me.

mDNS Browser is now marked as developer-approved on FlatPark and carries the blue shield on its app page (flatpark/flatpark#198). Thank you as well for taking the time to open flatpark/flatpark#197 yourself and for adding FlatPark to the README — the unbundled mdns-browser_linux_x64 binary is clearly the right payload here, and I'm glad you pointed it out.

One follow-up on your second point: the wrapper still sets WEBKIT_DISABLE_DMABUF_RENDERER=1. Once v1.15.2 is out and the pin moves to it, I'll drop that line and test the sandboxed build without it, so the app's own detection is what decides. If you'd rather I keep it until you've confirmed the quirk behaves inside the sandbox, just say the word.

If anything about the packaging ever looks wrong — a release the resolver misses, a permission that's too tight or too loose, anything — please ping me here or open an issue on flatpark/flatpark. I'm happy to fix it quickly.

**Reply by hrzlgnm** [reply]
Thanks, I'm not able to test that thoroughly as i don't have a machine with nvidia where flatpak is installed as I'm using an arch linux based disto and arch prefers AUR. So please test the v1.15.2 release without the flag before removing it.
